### ISSUE TITLE
Learning Rate reduce_lr_min_delta issue

### ISSUE THREAD

**Issue by realalikamali** [sciann/sciann#58]
Hi! I have recently come upon an issue when looking into learning rate reduction updates in my runs. This has to do with the learning_rate arguments in train()

the documentation on the website and the comments in the source code mention the following default values:

`"""...reduce_lr_after: patience to reduce learning rate or stop after certain missed epochs.
Defaulted to epochs max(10, epochs/10).
stop_lr_value: stop the training if learning rate goes lower than this value.
Defaulted to 1e-8.
reduce_lr_min_delta: min absolute change in total loss value that is considered a successful change.
Defaulted to 0.001. ..."""`

However, the actual default value in train() seems to be set to zero:

`reduce_lr_min_delta=0.,`

So with this setting, the learning rate should just automatically reduce every epochs/10 because the zero reduce_lr_min_delta will never be met.

But the surprising thing to me is that I have run models with these current defaults (reduce_lr_after=max(10, epochs/10), reduce_lr_min_delta=0.) and the learning rate does not necessarily decrease every epochs/10, and the behavior changes case by case: 
![image](https://user-images.githubusercontent.com/60515966/192039115-d8065ef6-4040-4279-86a2-279d0888e893.png)
![image](https://user-images.githubusercontent.com/60515966/192039241-c63aaba9-8418-489b-9e47-f3b69cd0ffe0.png)
![image](https://user-images.githubusercontent.com/60515966/192039340-2adff1d7-62bb-459f-9672-ef9c836bb17b.png)

I am wondering if there is any bug in the default setting of reduce_lr_min_delta in the library and if not, why does that definition lead to the example results I shared?

Thanks!

**Comment by realalikamali** [comment]
Well, after a closer look, I have found out that the train() function currently **does** default to a reduce_lr_min_delta of zero, so I believe the documentation/code comments need to be updated.

In my previous post, I made a mistake in interpreting how a reduce_lr_min_delta of zero works. Now I understand that with this reduce_lr_min_delta value, the learning rate scheduler keeps resetting its counter to zero after every epoch that the loss is smaller than the minimum loss up to that point. Otherwise, it adds to that counter. when the counter reaches the predefined patience value, the learning rate gets updated, the counter resets to zero, and the new minimum loss value is set to the current loss at that epoch.

Hope this helped clarify things if other people had a similar issue.
